### PR TITLE
DM-24658: Improve error reporting in connection string.

### PR DESCRIPTION
**Pull request by DinoBektesevic**
Stop silencing `DbAuthError` when cause is a YAML parsing issue. 
Reorganize dbAuth and connectionString test files, simplify and de-duplicate. 
Add test to ensure error propagates through ConnectionString class .

### CODE REVIEW

File: python/lsst/daf/butler/registry/_dbAuth.py [hunk 1]
```diff
@@ -79,6 +83,8 @@ def __init__(self, path=None, envVar=None, authList=None):
                 "{:o}".format(secretPath, mode))
 
         try:
+            if not os.path.isfile(secretPath):
+                raise DbAuthNotFoundError(f"DbAuth file not found: {secretPath}")
```
Comment (ktlim) [inline python/lsst/daf/butler/registry/_dbAuth.py]: This should be removed and the proper exception thrown at (new) line 78.

File: python/lsst/daf/butler/registry/connectionString.py [hunk 2]
```diff
@@ -91,13 +93,15 @@ def fromConfig(cls, registryConfig):
...
-            # re-raise permission error for safety
-            raise
-        except DbAuthError:
-            # credentials file doesn't exist or no match was found
+        except DbAuthNotFoundError:
+            # credentials file doesn't exist or no matches were found
             pass
+        except Exception:
+            # re-raise everything else
+            raise
```
Comment (ktlim) [inline python/lsst/daf/butler/registry/connectionString.py]: Is this necessary?  Leaving it out should cause the exception to propagate normally.

File: tests/config/dbAuth/registryConf2.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# minimal configuration required when using explicit 'host-port',
+# also tests '/' are placed correctly when not epxlicit in dbname
```
Comment (ktlim) [inline tests/config/dbAuth/registryConf2.yaml]:
```suggestion
# also tests '/' are placed correctly when not explicit in dbname
```

**Review by ktlim — APPROVED** [review]

File: python/lsst/daf/butler/registry/_dbAuth.py (hunk 1 above)
Comment (DinoBektesevic) [reply python/lsst/daf/butler/registry/_dbAuth.py]: Thanks! I guess I was blind.

**Pull request merged**